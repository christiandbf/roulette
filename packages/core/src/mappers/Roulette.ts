import Mapper from './Mapper';
import Roulette from '../domain/RouletteEntity';
import {
	RouletteRequestModel,
	RouletteResponseModel
} from '../models/Roulette';

const RouletteMapper: Mapper<
	Roulette,
	RouletteRequestModel,
	RouletteResponseModel
> = {
	toEntity: (request: RouletteRequestModel): Roulette => {
		return new Roulette({ name: request.name, id: request.id || undefined });
	},
	toModel: (roulette: Roulette): RouletteResponseModel => ({
		id: roulette.getId(),
		isOpen: roulette.getIsOpen(),
		name: roulette.getName()
	})
};

export default RouletteMapper;
