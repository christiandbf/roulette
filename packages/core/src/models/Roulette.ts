export interface RouletteRequestModel {
	id?: string;
	name: string;
}

export interface RouletteResponseModel {
	id: string;
	isOpen: boolean;
	name: string;
}
