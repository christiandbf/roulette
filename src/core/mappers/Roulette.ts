import Mapper from './Mapper';
import Roulette from '../domain/RouletteEntity';
import {
  RouletteRequestModel,
  RouletteResponseModel
} from '../models/Roulette';

const RouletteMapper: Mapper<
  Roulette,
  RouletteRequestModel,
  RouletteResponseModel
> = {
  toEntity: (request: RouletteRequestModel): Roulette => {
    return new Roulette({ name: request.name });
  },
  toModel: (roulette: Roulette): RouletteResponseModel => ({
    id: roulette.getId(),
    isOpen: roulette.getIsOpen(),
    name: roulette.getName()
  })
};

export default RouletteMapper;
