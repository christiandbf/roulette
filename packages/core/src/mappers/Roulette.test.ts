import Roulette from '../domain/RouletteEntity';
import RouletteMapper from './Roulette';

describe('Roulette mapper', () => {
	it('toEntity', () => {
		const entity = RouletteMapper.toEntity({
			name: 'Royal 007',
			id: 'ff6960b0-b015-11eb-8b4e-bd5c14d98229'
		});
		expect(entity).toMatchSnapshot();
	});

	it('toEntity creating new ID', () => {
		const entity = RouletteMapper.toEntity({
			name: 'Royal 007'
		});
		expect(entity.getId()).toBeTruthy();
	});

	it('toModel', () => {
		const roulette = new Roulette({
			name: 'Royal 007',
			id: 'ff6960b0-b015-11eb-8b4e-bd5c14d98229'
		});
		const model = RouletteMapper.toModel(roulette);
		expect(model).toMatchSnapshot();
	});
});
