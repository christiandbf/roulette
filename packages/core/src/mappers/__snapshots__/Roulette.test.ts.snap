// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Roulette mapper toEntity 1`] = `
Roulette {
  "MIN_CHARACTERS": 2,
  "UUID_REGEXP": /\\^\\[0-9a-f\\]\\{8\\}-\\[0-9a-f\\]\\{4\\}-\\[0-5\\]\\[0-9a-f\\]\\{3\\}-\\[089ab\\]\\[0-9a-f\\]\\{3\\}-\\[0-9a-f\\]\\{12\\}\\$/i,
  "id": "ff6960b0-b015-11eb-8b4e-bd5c14d98229",
  "isOpen": false,
  "name": "Royal 007",
  "props": Object {
    "id": "ff6960b0-b015-11eb-8b4e-bd5c14d98229",
    "name": "Royal 007",
  },
}
`;

exports[`Roulette mapper toModel 1`] = `
Object {
  "id": "ff6960b0-b015-11eb-8b4e-bd5c14d98229",
  "isOpen": false,
  "name": "Royal 007",
}
`;
